magstock/templates/regextra.html

```diff
@@ -13,7 +13,9 @@
             };
         {% endif %}
         $(function () {
-            $.field('amount_extra').parents('.form-group').hide();
+            if ($.field('amount_extra')) {
```

Comment:
This is the change that actually fixes the bugs in question. The transfer page is a unique case where we have the BADGE_TYPES window but not the amount_extra box, so we need an extra check to make sure it exists - otherwise, it creates a JS error, which breaks other JS on the page.